.github/workflows/codeql.yml

```diff
@@ -0,0 +1,63 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ 'master' ]
+  pull_request:
+    branches: [ 'master' ]
+  schedule:
+    - cron: '39 18 * * 0'
+
+env:
+  UBUNTU_DEP_BUILD: |
+    gcc g++ pkg-config systemd libevdev-dev libglib2.0-dev
+    libjson-glib-dev libsystemd-dev libudev-dev libunistring-dev
+    python3-dev python3-evdev swig python3-sphinx
+  UBUNTU_DEP_TEST: check python3-gi python3-lxml valgrind
+  PIP_PACKAGES: meson ninja libevdev pyudev pytest sphinx_rtd_theme
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-20.04
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp', 'python' ]
+        meson_args:
+          - ''
+          - '-Dbuildtype=plain'
+          - '-Dbuildtype=release'
+          - '-Ddocumentation=true'
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - uses: ./.github/actions/pkginstall
+      with:
+        apt: $UBUNTU_DEP_BUILD $UBUNTU_DEP_TEST
+        pip: $PIP_PACKAGES
+
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v2
+      with:
+        languages: ${{ matrix.language }}
```

Comment:
I wonder if instead of defining languages in the matrix we can simplify the workflow to just list them here, and not run twice the amount of workflows.

Reply:
Seems like we can! Great!